js/src/lib/interfaces/Libraries.ts

```diff
@@ -106,11 +106,41 @@ const asteroid = (model: ModelData) =>
 
 model = BaseModel.from_pretrained("${model.id}")`;
 
-const diffusers = (model: ModelData) =>
-	`from diffusers import DiffusionPipeline
+const diffusers_default = (model: ModelData) =>
+  `from diffusers import DiffusionPipeline
 
 pipeline = DiffusionPipeline.from_pretrained("${model.id}")`;
 
+const diffusers_controlnet = (model: ModelData) =>
+  `from diffusers import StableDiffusionControlNetPipeline, ControlNetModel
+
+controlnet = ControlNetModel.from_pretrained("${model.id}")
+pipeline = StableDiffusionControlNetPipeline.from_pretrained("${model.cardData?.base_model}", controlnet=controlnet)`;
+
+const diffusers_lora = (model: ModelData) =>
+  `from diffusers import DiffusionPipeline
+
+pipeline = DiffusionPipeline.from_pretrained("${model.cardData?.base_model}")
+pipeline.load_lora_weights("${model.id}")`;
+
+const diffusers_textual_inversion = (model: ModelData) =>
+  `from diffusers import DiffusionPipeline
+
+pipeline = DiffusionPipeline.from_pretrained("${model.cardData?.base_model}")
+pipeline.load_textual_inversion("${model.id}")`;
+
+const diffusers = (model: ModelData) => {
+  if (model.tags?.includes("controlnet")) {
```

Comment:
```suggestion
  if (model.tags?.includes("controlnet") && model.cardData?.base_model) {
```
or otherwise add this check everywhere?

Reply:
Added a function now: `get_base_diffusers_model` which will return "fill-in-base-model" if nothing is specified